content/blog/2018-09-04-birds-taxo-traits.md

```diff
@@ -0,0 +1,541 @@
+                                    "figure-markdown_github", "taxo.gif"),
+               delay = 3,
+               width = 700, height = 700,
+               progress = FALSE)
+```
+
+    ## [1] "/img/blog-images/2018-09-04-birds-taxo-traits/taxo.gif"
+
+``` r
+knitr::include_graphics(file.path("2018-09-04-birds-taxo-traits_files",
+                                    "figure-markdown_github", "taxo.gif")) 
+```
+
+![animated tree with species names and order
+silhouette](/img/blog-images/2018-09-04-birds-taxo-traits/taxo.gif)
+
+This gif shows many species names and order giving us a feeling for what
+we might encounter in the county of Constance, but it lacks quantitative
+information about the species. It’d be interesting to create trees such
+as the ones of the [`metacorder`
+package](https://github.com/grunwaldlab/metacoder) to reflect abudance,
+possibly depending on the very local area (distance to watery area) or
+season, potentially using the `taxize::downstream` function to get all
+families in each order, even families not present in our occurrence
+dataset. This idea is beyond the scope of this post. What *is* in scope,
+now, is trying to get trait information for the species.
+
+### Getting trait information
+
+In ecology, *traits* are characteristics of organisms such as habitat,
+body size, threats, etc. It’s a whole bunch of data you can get for free
+based on species scientific names, from different data providers. The
+[`traits` package](https://github.com/ropensci/traits), part of the
+rOpenSci’s suite, is an interface to various sources of traits data. In
+this section, we shall use data from BirdLife International: habitat and
+threats.
+
+The different functions of `traits` have prefixes indicating with which
+data source they interact. Here we shall use `traits::birdlife_habitat`
+and `traits::birdlife_threats`. To get access to the data available for
+each species, we first need to get its IUCN ID using the `taxize`
+package (or the `rredlist` package that it wraps)
+
+``` r
+species <-  unique(ebd$scientific_name)
+
+get_info <- function(species){
+  message(species)
+  Sys.sleep(1)
+  iucn_id <- taxize::iucn_id(species)
+  if(!is.na(iucn_id)){
+    habitat <- traits::birdlife_habitat(iucn_id)
+    threats <- traits::birdlife_threats(iucn_id)
+  }else{
+    habitat <- NULL
+    threats <- NULL
+  }
+ 
+  
+  list(habitat = habitat,
+       threats = threats)
+}
+
+species_info <- purrr::map(species,
+                           get_info)
+
+names(species_info) <- species
+save(species_info, file = file.path("taxo", "species_info.RData"))
+```
+
+The script above isn’t the smartest since it doesn’t retain information
+about the species names. Not too bad here because I want to get a
+general idea of birds’ habitats and threats over the county.
+
+``` r
+habitat <- purrr::map_df(species_info, "habitat")
+```
+
+Out of 0, 203 are represented in this dataset.
```

Comment:
> Out of 0, 203 are represented in this dataset.

i'm not sure this makes sense or maybe i'm reading it wrong?

Reply:
Oops buggy inline code